src/HydraAuction/OnChain/FeeEscrow.hs

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS_GHC -fno-specialise #-}
+
+module HydraAuction.OnChain.FeeEscrow (mkFeeEscrowValidator) where
+
+-- Prelude imports
+import PlutusTx.Prelude
+
+-- import Prelude (quot)
+
+-- Plutus imports
+import Plutus.V1.Ledger.Address (pubKeyHashAddress, scriptHashAddress)
+import Plutus.V1.Ledger.Value (valueOf)
+import Plutus.V2.Ledger.Api (TxInfo (..), TxOut (..), Value, adaSymbol, adaToken, scriptContextTxInfo, txInInfoResolved)
+import Plutus.V2.Ledger.Contexts (ScriptContext, ownHash)
+
+-- Hydra auction imports
+import HydraAuction.Types (
+  AuctionTerms (..),
+  FeeEscrowDatum,
+  FeeEscrowRedeemer (..),
+ )
+import HydraAuctionUtils.Plutus (
+  byAddress,
+  nothingForged,
+ )
+
+{-# INLINEABLE mkFeeEscrowValidator #-}
+mkFeeEscrowValidator :: AuctionTerms -> FeeEscrowDatum -> FeeEscrowRedeemer -> ScriptContext -> Bool
+mkFeeEscrowValidator terms () DistributeFees context =
+  -- Every delegate is payed at least the expected proportion of fee
+  all receivesProportionOfFee (delegates terms)
+    -- No tokens are minted or burned.
+    && nothingForged info
+  where
+    info :: TxInfo
+    info = scriptContextTxInfo context
+
+    outputs :: [TxOut]
+    outputs = txInfoOutputs info
+
+    adaValueOf v = valueOf v adaSymbol adaToken
+
+    expectedValuePerDelegate :: Integer
+    expectedValuePerDelegate = adaValueOf singleFeeInputValue `quotient` length (delegates terms)
```

Comment:
This is just `auctionFeePerDelegate`. This cannot have a reminder, cuz `Escrow` ensures that `calculateTotalFee terms` is emitted. If it was not so amount should be checked anyway.

 So no need for all this stuff.